crates/solver/src/interactions/allowances.rs

```diff
@@ -209,7 +215,7 @@ where
     let mut allowances = HashMap::new();
     for (spender, token, allowance) in results {
         let allowance = match allowance {
-            Ok(value) => U256::from(value.0.as_slice()),
+            Ok(allowance) => U256::from_be_slice(&allowance.0),
```

Comment:
If the future that produces this value uses `Erc20Instance::allowance()` instead of `web3.eth().call()` `alloy` would do the right conversion for us and we wouldn't have to guess the endinanness here.

Reply:
I converted this piece, but the issue is that it would not pass by the web3 mock. This is just a "I forgot what the original code was" moment, I'll fix the mock issue right after this PR